components/Nav/MobileNav.tsx

```diff
@@ -0,0 +1,145 @@
+import { Disclosure, Transition } from "@headlessui/react";
+import { signIn } from "next-auth/react";
+import Link from "next/link";
+import { FunctionComponent } from "react";
+import { navigation, subNav, userSubNav } from "../../config/site_settings";
+import { UserNavigationItem } from "@/types/types";
+import { Session } from "next-auth";
+
+function classNames(...classes: string[]) {
+  return classes.filter(Boolean).join(" ");
+}
+
+interface MobileNavProps {
+  session: Session | null;
+  userNavigation: UserNavigationItem[];
+}
+
+const MobileNav: FunctionComponent<MobileNavProps> = ({
+  session,
+  userNavigation,
+}) => {
+  return (
+    <Transition
+      enter="transition-transform duration-200"
+      enterFrom="transform -translate-y-full opacity-0"
+      enterTo="transform translate-y-0 opacity-100"
+      className="absolute w-screen z-10 bg-neutral-100 dark:bg-black"
+    >
+      <Disclosure.Panel className="md:hidden relative z-10 border-b-2 border-black dark:border-white">
+        <div className="px-2 pt-2 pb-3 space-y-1">
+          {navigation.map((item) =>
```

Comment:
great job on this! I found it a little hard to read through. I am wondering if we can break some of these nested ternaries into separate components for readability? 

Something like

`<NavItem item={item} />`

Where NavItem is a new component that handles the different types of navigation items. This way, the main component stays a bit cleaner